protocol/daemons/liquidation/client/client.go

```diff
@@ -65,262 +80,56 @@ func Start(
 	liquidationServiceClient := api.NewLiquidationServiceClient(daemonConn)
 
 	ticker := time.NewTicker(time.Duration(flags.Liquidation.LoopDelayMs) * time.Millisecond)
-	for ; true; <-ticker.C {
-		if err := RunLiquidationDaemonTaskLoop(
-			ctx,
-			flags.Liquidation,
-			subaccountQueryClient,
-			clobQueryClient,
-			liquidationServiceClient,
-		); err != nil {
-			// TODO(DEC-947): Move daemon shutdown to application.
-			logger.Error("Liquidations daemon returned error", "error", err)
-		}
-	}
-
-	return nil
-}
+	done := make(chan bool)
 
-// RunLiquidationDaemonTaskLoop contains the logic to communicate with various gRPC services
-// to find the liquidatable subaccount ids.
-func RunLiquidationDaemonTaskLoop(
-	ctx context.Context,
-	liqFlags flags.LiquidationFlags,
-	subaccountQueryClient satypes.QueryClient,
-	clobQueryClient clobtypes.QueryClient,
-	liquidationServiceClient api.LiquidationServiceClient,
-) error {
-	defer telemetry.ModuleMeasureSince(
-		metrics.LiquidationDaemon,
-		time.Now(),
-		metrics.MainTaskLoop,
-		metrics.Latency,
-	)
-
-	// 1. Fetch all subaccounts from query service.
-	subaccounts, err := GetAllSubaccounts(
+	s := &SubTaskRunnerImpl{}
+	StartLiquidationsDaemonTaskLoop(
 		ctx,
+		c,
+		s,
+		flags,
+		ticker,
+		done,
 		subaccountQueryClient,
-		liqFlags.SubaccountPageLimit,
-	)
-	if err != nil {
-		return err
-	}
-
-	// 2. Check collateralization statuses of subaccounts with at least one open position.
-	liquidatableSubaccountIds, err := GetLiquidatableSubaccountIds(
-		ctx,
 		clobQueryClient,
-		liqFlags,
-		subaccounts,
-	)
-	if err != nil {
-		return err
-	}
-
-	// 3. Send the list of liquidatable subaccount ids to the daemon server.
-	err = SendLiquidatableSubaccountIds(
-		ctx,
 		liquidationServiceClient,
-		liquidatableSubaccountIds,
+		logger,
 	)
-	if err != nil {
-		return err
-	}
 
 	return nil
 }
 
-// GetAllSubaccounts queries a gRPC server and returns a list of subaccounts and
-// their balances and open positions.
-func GetAllSubaccounts(
+// StartLiquidationsDaemonTaskLoop contains the logic to periodically run the liquidations daemon task.
+func StartLiquidationsDaemonTaskLoop(
 	ctx context.Context,
-	client satypes.QueryClient,
-	limit uint64,
-) (
-	subaccounts []satypes.Subaccount,
-	err error,
+	client *Client,
+	s SubTaskRunner,
+	flags flags.DaemonFlags,
+	ticker *time.Ticker,
+	done <-chan bool,
```

Comment:
why do we need this? and what triggers this?

Reply:
I think this design was originally to help us to detect when to halt the daemon based on when the stop channel is closed - in the daemon, we don't send a value to `stop`, we just close it and it emits a single nil value. But actually there is a syntax that lets you detect whether a channel is closed when you read a value:

`_, ok := <- ticker.C`

So it looks like I can refactor both here and the pricefeed daemon to get rid of the stop channel.

Reply:
when a ticker stops ticking, it does not close it's tick channel, so we added another channel to directly signal the stop so that we can halt the daemons. done is triggered by whoever wants the infinite loop to halt - in this case, the test cases, but in the future, a Stop method on the daemon could also trigger the done channel.

Reply:
```suggestion
	stop <-chan bool,
```